### PR TITLE
haskell-language-server: 0.8.0 -> 0.9.0

### PR DESCRIPTION
**Pull request by Lucus16**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cdepillabout** [comment]
Ping @maralorn and @GuillaumeDesforges

**Review by maralorn — CHANGES_REQUESTED** [review]
Hey @Lucus16,

thank you for your contribution. It’s obvious from the PR that you have put some work into it and I appreciate it. I have commented on a few details, and there is also one big thing.

Our practice here in nixpkgs is to always use the package versions out of hackage. The fact that we didn‘t do this for haskell-language-server is only caused by the fact, that hls wasn’t released to hackage at the time of the 0.8.0.0 release.
But now it is. So at this point the aim should be this:
- [ ] Remove all hls related packages from `non-hackage-packages` and remove their files.
- [ ] Use the hackage version of hls and set the cabal flags accordingly. My aim is to enable all-formaters and all-plugins.
- [ ] See if this works and overrides to dependencies were necessary. Worst case scenario: We need to provide some dependencies as non-hackage-packages, Add this point I wouldn‘t be surprised if bug reports for upstream come up.
- [ ] Urge upstream to release 0.9.0 to hackage. (There is no Haskell application in nixpkgs for which we manually maintain a newer version than the released-to-hackage one. I don‘t think we should make an exception here.)

If you don‘t care to try this I fully understand. But I would be exited if you do. My suggestion would be to leave this PR open as a fallback in case my plan doesn‘t work and try a new PR with the changes I outlined here.

I am sorry that your PR falls exactly into this transition phase, but we really should do this to have a more future-proof setup which in the end will make PRs like this completely unnecessary.

File: pkgs/development/haskell-modules/non-hackage-packages.nix [hunk 1]
```diff
@@ -27,6 +27,8 @@ self: super: {
   hls-retrie-plugin = self.callPackage ../tools/haskell/haskell-language-server/hls-retrie-plugin.nix { };
   hls-class-plugin = self.callPackage ../tools/haskell/haskell-language-server/hls-class-plugin.nix { };
   hls-eval-plugin = self.callPackage ../tools/haskell/haskell-language-server/hls-eval-plugin.nix { };
+  ghcide = self.callPackage ../tools/haskell/haskell-language-server/ghcide.nix { };
+  hls-plugin-api = self.callPackage ../tools/haskell/haskell-language-server/hls-plugin-api.nix { };
```
[the remark this answers is not in the document]
Comment (yorickvP) [reply pkgs/development/haskell-modules/non-hackage-packages.nix:31]: The ghcide-0.7.2.0 from https://github.com/haskell/haskell-language-server/tree/0.9.0/ghcide is substantially different from the one on hackage.

[the remark this answers is not in the document]
Comment (maralorn) [reply pkgs/development/haskell-modules/non-hackage-packages.nix:31]: Ugh, that‘s sad.

Well, in the big picture it doesn‘t make much of a difference either we take everything from hackage or from github it seems.

**Comment by GuillaumeDesforges** [comment]
@maralorn said it all :) we should transition to hackage now

**Comment by cdepillabout** [comment]
@maralorn Should this be closed since #112063 was already merged in?

**Comment by Lucus16** [comment]
Yes, and thanks to everyone that helped!

**Pull request closed without merging**